consensus/src/experimental/pipeline_phase.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use futures::{
+    channel::mpsc::{UnboundedReceiver, UnboundedSender},
+    SinkExt, StreamExt,
+};
+
+#[async_trait]
+pub trait StatelessPipeline<T, V> {
```

Comment:
we can directly have StatelessPipeline<T, V>: Send + Sync here to avoid adding it in every box